src/tests/ftest/control/version.py

```diff
@@ -41,66 +50,23 @@ def test_version(self):
             self.log.info("RPM version = %s", rpm_version)
 
         # Get dmg version.
-        dmg_cmd = self.get_dmg_command()
-        output = dmg_cmd.version().stdout.decode("utf-8")
-
-        # Verify that "dmg version" is in the output.
-        if "dmg version" not in output:
-            errors.append("dmg version is not in the output! {}".format(output))
-
-        result = re.findall(r"dmg version ([\d.]+)", output)
-        if not result:
-            errors.append("Failed to obtain dmg version! {}".format(output))
-        else:
-            dmg_version = result[0]
-            self.log.info("dmg version = %s", dmg_version)
+        dmg_version = self.get_dmg_command().version()["response"]["version"]
+        self.log.info("dmg version = %s", dmg_version)
 
         # Get daos version.
-        daos_cmd = self.get_daos_command()
-        output = daos_cmd.version().stdout.decode("utf-8")
-
-        # Verify that "daos version" is in the output.
-        if "daos version" not in output:
-            errors.append("daos version is not in the output! {}".format(output))
-
-        result = re.findall(r"daos version ([\d.]+)", output)
-        if not result:
-            errors.append("Failed to obtain daos version! {}".format(output))
-        else:
-            daos_version = result[0]
-            self.log.info("daos version = %s", daos_version)
+        daos_version = self.get_daos_command().version()["response"]["version"]
+        self.log.info("daos version = %s", daos_version)
 
         # Get daos_agent version.
-        daos_agent_cmd = "daos_agent version"
+        daos_agent_cmd = "daos_agent --json version"
         output = run_pcmd(hosts=self.hostlist_servers, command=daos_agent_cmd)
-        stdout = output[0]["stdout"][0]
-
-        # Verify that "DAOS Agent" is in the output.
-        if "DAOS Agent" not in stdout:
-            errors.append("DAOS Agent is not in the output! {}".format(stdout))
-
-        result = re.findall(r"DAOS Agent v([\d.]+)", stdout)
-        if not result:
-            errors.append("Failed to obtain daos_agent version! {}".format(output))
-        else:
-            daos_agent_version = result[0]
-            self.log.info("daos_agent version = %s", daos_agent_version)
+        daos_agent_version = json.loads("".join(output[0]["stdout"]))["response"]["version"]
```

Comment:
The field `exit_status` should probably be checked before accessing the `output` one.